akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -244,7 +286,31 @@ object ShardCoordinator {
      */
     def rebalance(
         currentShardAllocations: java.util.Map[ActorRef, immutable.IndexedSeq[String]],
-        rebalanceInProgress: java.util.Set[String]): Future[java.util.Set[String]]
+        rebalanceInProgress: java.util.Set[String]): Future[java.util.Set[String]] = {
+      import scala.jdk.FutureConverters.CompletionStageOps
+      rebalanceShards(currentShardAllocations, rebalanceInProgress).asScala
+    }
```

Comment:
Same with the deprecation (and you may need to add some nowarn:s in the new methods calling this as well)